litellm/main.py

```diff
@@ -910,7 +910,7 @@ def completion(
             llm_provider=custom_llm_provider, dynamic_api_key=api_key
         )  # get the api key from the environment if required for the model
 
-        if dynamic_api_key is not None:
+        if dynamic_api_key is not None and api_key is None:
```

Comment:
@Raymond1415926 this is the wrong place for the fix. 

It should be in get_llm_provider - https://github.com/BerriAI/litellm/blob/9e07af6c0245ba19696aea27b6a94ee1f587fe97/litellm/utils.py#L4274

Reply:
Can I make it global? Or it has to be case by case?

Other providers, by providing the same get_secret logic, will result in the same issue where input api_key never overrides the environmental variables.

Reply:
i think a global approach is fine